Samples/ProjectTracker/ProjectTracker.Ui.Blazor.Server/Pages/ProjectEdit.razor

```diff
@@ -0,0 +1,229 @@
+</p>
+
+@if (vm.Model == null)
+{
+  <p>Loading...</p>
+}
+else 
+{ 
+  <table class="table">
+    <thead>
+      <tr>
+        <th></th>
+        <th></th>
+      </tr>
+    </thead>
+    <tbody>
+      @if (vm.Model.CanReadProperty(nameof(vm.Model.Id)))
+      {
+        <tr>
+          <td>Id</td>
+          <td>@vm.Model.Id</td>
+        </tr>
+      }
+      @if (vm.GetPropertyInfo<string>(nameof(vm.Model.Name)).CanRead)
+      {
+        <tr>
+          <td>@(vm.GetPropertyInfo<string>(nameof(vm.Model.Name)).FriendlyName)</td>
+          <td>
+            <TextInput Property="@(vm.GetPropertyInfo<string>(nameof(vm.Model.Name)))" />
+          </td>
+        </tr>
+      }
+      @if (vm.GetPropertyInfo<string>(nameof(vm.Model.Description)).CanRead)
+      {
+        <tr>
+          <td>@(vm.GetPropertyInfo<string>(nameof(vm.Model.Description)).FriendlyName)</td>
+          <td>
+            <TextArea rows="5" Property="@(vm.GetPropertyInfo<string>(nameof(vm.Model.Description)))" />
+          </td>
+        </tr>
+      }
+      @if (vm.GetPropertyInfo<DateTime?>(nameof(vm.Model.Started)).CanRead)
+      {
+        <tr>
+          <td>@(vm.GetPropertyInfo<DateTime?>(nameof(vm.Model.Started)).FriendlyName)</td>
+          <td>
+            <DateInput Property="@(vm.GetPropertyInfo<DateTime?>(nameof(vm.Model.Started)))" />
+          </td>
+        </tr>
+      }
+      @if (vm.GetPropertyInfo<DateTime?>(nameof(vm.Model.Ended)).CanRead)
+      {
+        <tr>
+          <td>@(vm.GetPropertyInfo<DateTime?>(nameof(vm.Model.Ended)).FriendlyName)</td>
+          <td>
+            <DateInput Property="@(vm.GetPropertyInfo<DateTime?>(nameof(vm.Model.Ended)))" EmptyIsMin="false" />
+          </td>
+        </tr>
+      }
+    </tbody>
+  </table>
+  <div style="background-color:lightgray">
+    @if (viewMode == SubViewModes.Details)
+    {
+      <a href="javascript: void(0);" @onclick="ResetResourcesView">Cancel assignment</a>
+      <table class="table">
+        <thead>
+          <tr>
+            <th>Name</th>
+            <th>Role</th>
+            <th></th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr>
+            <td>@selectedResource.FirstName @selectedResource.LastName</td>
+            <td>
+              <select @bind="selectedResource.Role">
+                @foreach (var item in ProjectTracker.Library.RoleList.GetCachedList())
+                {
+                  <option value="@item.Key">@item.Value</option>
+                }
+              </select>
+            </td>
+            <td><a href="javascript: void(0);" @onclick="() => AddResource()">Assign</a></td>
+          </tr>
+        </tbody>
+      </table>
+    }
+    else if (viewMode == SubViewModes.Select)
+    {
+      @if (_resourceList == null)
+      {
+        <p>Loading resource list...</p>
+      }
+      else
+      {
+        <a href="javascript: void(0);" @onclick="ResetResourcesView">Cancel assignment</a>
```

Comment:
I'm not sure, but might the new preventDefault feature added in .Net Core 3.1 avoid the need for the javascript in anchor tags?

Reply:
I hope so! I did try that feature last week and it didn't work, which is why I'm still using the javascript. I should probably try again now that they've released a new preview.